compiler/commands.nim

```diff
@@ -271,6 +271,7 @@ proc testCompileOption*(switch: string, info: TLineInfo): bool =
   of "patterns": result = contains(gOptions, optPatterns)
   of "experimental": result = gExperimentalMode
   of "excessivestacktrace": result = contains(gGlobalOptions, optExcessiveStackTrace)
+  of "sourcemap": result = gOptions * {optLineDir, optSourceMap} == {optLineDir, optSourceMap}
```

Comment:
Why would I need to check for that in a ``when compileOption``? Remove this line please.

Reply:
well you might want to debug log something only if a certain debugging option is not turned on(to help you debug it in this case)

probably not useful indeed, I'll remove it

Reply:
if this is in testCompileOption, ok